vernac/declare.ml

```diff
@@ -642,14 +642,31 @@ let declare_assumption ~name ~scope ~hook ~impargs ~uctx pe =
   dref
 
 (* Preparing proof entries *)
+let error_unresolved_evars env sigma t evars =
+  let pr_unresolved_evar e =
+    hov 2 (str"- " ++ Printer.pr_existential_key sigma e ++  str ": " ++
+      Himsg.explain_pretype_error env sigma
```

Comment:
You could expose explain_unsolvable_implicit for this. Not sure if that's better.

Reply:
Hum, either I stay like this or we craft a new error that take the evar set and the term and that is printed as I do.
I mean, move all of the new code to Himsg.

Exposing more API for this use case seems a bit too ad-hoc to me.